nntrainer/graph/network_graph.cpp

```diff
@@ -268,33 +267,44 @@ int NetworkGraph::realizeActivationType(Layer &current) {
   current.output_layers.clear();
   current.output_layers.push_back(layer->getName());
 
-  addLayerNode(layer);
-
   updateNameInLayers(current.getName(), layer->getName());
+  addLayerNode(layer);
 
   return status;
 }
 
 int NetworkGraph::addLossLayer(const LossType loss_type) {
+
   int status = ML_ERROR_NONE;
+
+  if (Sorted.back().layer->getType() == LossLayer::type)
+    return status;
+
+  if (loss_type == LossType::LOSS_NONE) {
+    status = ML_ERROR_INVALID_PARAMETER;
+    NN_RETURN_STATUS();
+  }
+
   LossType updated_loss_type = loss_type;
   if (num_node == 0) {
     status = ML_ERROR_INVALID_PARAMETER;
     NN_RETURN_STATUS();
   }
 
+  LayerNode last_node = Sorted.back();
   if (updated_loss_type == LossType::LOSS_ENTROPY) {
-    if (getLayerNode(num_node - 1).layer->getType() != "activation") {
+    if (last_node.layer->getType() != "activation") {
```

Comment:
istrequal here?

Reply:
This is different from the layer name. what is our policy for this?

Reply:
I checked, type and layer names are case-insensitive.

Reply:
But we can make layer names as case-sensitive, leaving as an issue I guess.

Reply:
i will let #1024 handle it.